### ISSUE TITLE
Anon-Connection-Wizard does not start in sys-whonix (whonix-gw 15)

### ISSUE THREAD

**Issue by 0brand** [tasket/Qubes-VM-hardening#29]
When `sys-whonix` is started for the first time `Anon-Conection-Wizard` is supposed to automaicaly start, and walk users through Tor setup and configuration. However, unlike in previous Whonix versions {13,14} when `vm-boot-protect` is configured in `sys-whonix` (`whonix-gw-15`) this does not happen. 

A workaround is to either start Anon-Connecton-Wizare manually _or_ run `whonixset up` to configure Tor and populate /var/lib/tor (with Tor State). This file is persistent across `sys-whonix` restarts. 

This is not due to Whonix files having been removed from /rw . I should have explained that better here: https://github.com/tasket/Qubes-VM-hardening/issues/18#issuecomment-510007431

**Comment by tasket** [comment]
Yes, I'm having trouble with it also on whonix 14. I'll have to research this further.

**Comment by adrelanos** [comment]
/etc/xdg/autostart/whonix-setup-wizard.desktop starts
/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start

What happens when starting /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start manually?

    /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start

----

qubes-whonix package file /usr/lib/qubes-bind-dirs.d/40_qubes-whonix.conf defines among others the following folder to be added to bind-dirs

    /var/cache/whonix-setup-wizard

Clearing /rw might wipe these files also.

----

The qubes-whonix package also ships the following files:

```
ls -la var/cache/whonix-setup-wizard/status-files/
total 8
drwxrwxrwx 2 user user 4096 Jun 30 04:40 .
drwxrwxrwx 3 user user 4096 Sep 30  2015 ..
-rw-r--r-- 1 user user    0 Jun 30 04:40 disclaimer.skip
-rw-r--r-- 1 user user    0 Jun 30 04:40 finish_page.skip
-rw-r--r-- 1 user user    0 Jun 30 04:40 first_use_check.skip
-rw-r--r-- 1 user user    0 Jun 30 04:40 whonix_repository.skip
```

Wondering if that could be the cause.
It wouldn't be hard to move these status skip files to a more appropriate folder which doesn't depend on bind-dirs.

**Comment by 0brand** [comment]
>What happens when starting /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start manually?
```
user@host:~$ sudo /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start
missing_disablenetwork_line
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
INFO: No page needs showing.
```
`Anon-Connection-Wizard` starts and Tor connects.

**Comment by adrelanos** [comment]
0brand:
>> What happens when starting /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start manually?
> ```
> user@host:~$ sudo /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start
> missing_disablenetwork_line
> QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
> INFO: No page needs showing.
> ```
> `Anon-Connection-Wizard` starts and Tor connects.


This is at the very first start of the VM? (Or from "simulated first
start" (if you reset the private image somehow)?)

If yes, then that's strange. Anon-Connection-Wizard (ACW) does not
autostart but when you run
/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start it works as
intended?

**Comment by 0brand** [comment]
>This is at the very first start of the VM? (Or from "simulated first start" (if you reset the private image somehow)?)

That was the simulated start. I confirmed by creating a fresh `sys-whonix` VM. Then,

`[user@dom0 ~] qvm-run sys-whonix xfce4-terminal`

Starts a terminal but `Anon-Connection-Wizard` does not start.

>If yes, then that's strange. Anon-Connection-Wizard (ACW) does not auto-start but when you run /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start it works as intended?

Yes, very strange. Just tried again to be sure. Same result. Anon-Connection-Wizard does not auto-start, running simulated start does the trick.

**Comment by adrelanos** [comment]
0brand:
>> This is at the very first start of the VM? (Or from "simulated first start" (if you reset the private image somehow)?)
> 
> That was the simulated start.


"simulated first start": I mean with a new private image. Just to make
sure we talk about the same thing.

Running /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start
manually is also somewhat simulated but let's call that manual start or
something to avoid confusion.

> I confirmed by creating a fresh `sys-whonix` VM. Then,
> 
> `[user@dom0 ~] qvm-run sys-whonix xfce4-terminal`
> 
> Starts a terminal but `Anon-Connection-Wizard` does not start.
> 
>> If yes, then that's strange. Anon-Connection-Wizard (ACW) does not auto-start but when you run /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start it works as intended?
> 
> Yes, very strange. Just tried again to be sure. Same result. Anon-Connection-Wizard does not auto-start, running simulated start does the trick.


So the first autostarted (if it autostarts hopefully) run of and
subsequent manual start of
/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start produces
different results.

Therefore we need to look "inside"
/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start to see what is
happening. I've added some output to make debugging easier. In future
versions output of
/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start will be
prefixed with
"/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start:" to make
searching the logs easier. Version you're having is not having this
usability feature yet and outputs only what it is actually noticing.

Luckily output of
/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start (when auto
started) will be stored in file:

~/.xsession-errors


Could you please have a look at
/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start as you
currently probably have?

https://github.com/Whonix/whonix-setup-wizard/blob/9a20891f82dc49fc7fb0f18e2522bc7e86421fd1/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start

Check for any line that includes

print

These are the possible outputs by
/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start ending up in
~/.xsession-errors

Alternatively post the whole ~/.xsession-errors here after a start of a
new sys-whonix.


Also, does

    whonixcheck --verbose

say something interesting?

Maybe some systemd unit is failing in this configuration?

    sudo systemctl --failed list-units

(whonixcheck --verbose should cover this.)


If someone speaking python... Do you see something in function
tor_status that could throw a python exception in this configuration?

https://github.com/Whonix/anon-connection-wizard/blob/master/usr/lib/python3/dist-packages/anon_connection_wizard/tor_status.py

**Comment by 0brand** [comment]
>> This is at the very first start of the VM? (Or from "simulated first start" (if you reset the private image somehow)?)
>
> That was the simulated start.
"simulated first start": I mean with a new private image. Just to make
sure we talk about the same thing.

 Yes I use a new private image for every test.

>Running /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start manually is also somewhat simulated but let's call that manual start or something to avoid confusion.

OK manual start it is ;)

> Could you please have a look at /usr/lib/whonix-setup-wizard/whonixsetup_check_for_start as you currently probably have? https://github.com/Whonix/whonix-setup-wizard/blob/9a20891f82dc49fc7fb0f18e2522bc7e86421fd1/usr/lib/whonix-setup-wizard/whonixsetup_check_for_start Check for any line that includes print

**Comment by 0brand** [comment]
I somehow managed to close the issue
